test/jdk/java/lang/Float/Binary16ConversionNaN.java

```diff
@@ -24,21 +24,27 @@
 /*
  * @test
  * @bug 8289551
+ * @library /test/lib
```

Comment:
If we want to simply not run this test on x87, that could be done using jtreg @requires tags. For example

* @requires vm.bits == "64"

if this is only an issue for 32-bit x86 or something based on requiring os.arch values.

The companion test [Binary16Conversion.java](https://github.com/openjdk/jdk/blob/master/test/jdk/java/lang/Float/Binary16Conversion.java) tests the NaN handling contract required by the specification.

Reply:
Oh, I did not realize other tests verify `NaN` works. I tried to salvage this test thinking we still need to test at least some `NaN`-s. Since there is no apparent need for that, let's just exclude the test on x87. See new commit.

Reply:
Right; anticipating the possibility of platform-specific NaN handling, I split out the bit-specific NaN tests into a separate file so just those NaN bit could have @requires tags while keeping most of the float16 conversion tests running on all platforms :-)